controller/proxy-injector/webhook.go

```diff
@@ -79,6 +78,13 @@ func Inject(api *k8s.API,
 	}
 	proxyInjectionAdmissionRequests.With(admissionRequestLabels(ownerKind, request.Namespace, report.InjectAnnotationAt, configLabels)).Inc()
 
+	for k, v := range nsAnnotations {
+		if k == pkgK8s.ProxyInjectAnnotation && v != pkgK8s.ProxyInjectEnabled {
+			log.Warnf("invalid value detected for annotation %s : %s", k, v)
+			recorder.Eventf(*parent, v1.EventTypeNormal, eventTypeSkipped, "Invalid value detected for annotation %s: %s", k, v)
```

Comment:
Please note that `parent` might be null here, so you also need to check for that.